### ISSUE TITLE
--max_fuzzy_junction=0

### ISSUE THREAD

**Issue by abayega** [Magdoll/cDNA_Cupcake#34]
Hi,

is it okay to run collapse_isoforms_by_sam.py with --max_fuzzy_junction set to 0 (zero)?

Thanks
A

**Comment by Magdoll** [comment]
Yes, you can run it with `--max_fuzzy_junction=0`. You may notice that there are more unique transcripts being declared at the end of the collapse step, even transcripts where they only differ 1 or 2 bp at the junctions, as residual errors in HQ isoforms, though rare, can occasionally cause alignment differences.

You can use the [summrize scripts here](https://github.com/Magdoll/cDNA_Cupcake/wiki/Summarizing-and-Scrubbing-Splice-Junctions-Across-Samples) to look at the differences of collapse results later.

**Comment by abayega** [comment]
Thank you Magdoll, is there any hope of making this script general so that it can be used for any full-length cDNA seq? I realised that when --max_fuzzy_junction>0 the collapse_isoforms_by_sam.py script starts to prob the read names and these will have to be PacBio format otherwise it throws a list index out of range error.

**Comment by Magdoll** [comment]
Hi @abayega ,
I just answered this question in [Cogent issue 13](https://github.com/Magdoll/Cogent/issues/13#issuecomment-391218545). 

You can hack the sequence ID using the short python snippet provided there. 

I'll consider allowing for generic IDs for the collapse script in the future.

--Liz

**Comment by abayega** [comment]
Great, thank you for the help. I will try it out...

**Comment by abayega** [comment]
Ooops, I

**Comment by abayega** [comment]
Sorry, I meant to just comment, you can close this whenever you want
